src/gt_pro.cpp

```diff
@@ -233,6 +239,11 @@ bool kmer_lookup_work(LmerRange* lmer_index, uint64_t* mmer_bloom, uint32_t* kme
 								const auto db_kmer = high_bits | low_bits;
 								assert(lmer == (db_kmer >> M2));
 								if (kmer == db_kmer) {
+									// The set of kmers that covers this SNP within a given read cannot conflict.
+									// They will all map to the same virtual SNP.  Therefore, we can use the snp_id
+									// (which is really a virtual snp id) as a proxy for the real snp id.  Even
+									// though in general the mapping of virtual to real snp id is many to one,
+									// within each read it is always 1:1.
```

Comment:
The statement in this comment is incorrect, and indeed we have a minor bug here that would result in slightly overcounting SNPs that have virtual counterparts (i.e. SNPs that occur in close proximity to other SNPs). 

Not to worry, this bug is fixed in one of the subsequent PRs.